### ISSUE TITLE
Error dumping metadata

### ISSUE THREAD

**Issue by KiwiKilian** [kontena/mongo-backup#11]
Sorry for bothering with a problem that might not be directly caused by this image, but I can't find any information on the error given. I had a connection error before, but I managed to fix it. I'm very new to docker and therefore not sure if my environment variables are correctly and might be causing a problem? The very little docs of this image made me guess most stuff from source code.
The following are the logs of the mongo-backup image:
```
docker logs mongo-backup
=== BACKUP WORKER ===
[2017/04/14 08:09:06][error] Model::Error: Backup for MongoDB backup (mongo_backup) Failed!
[2017/04/14 08:09:06][error] --- Wrapped Exception ---
[2017/04/14 08:09:06][error] Utilities::Error: 'mongodump' failed with exit status: 1
[2017/04/14 08:09:06][error]   STDOUT Messages: None
[2017/04/14 08:09:06][error]   STDERR Messages: 
[2017/04/14 08:09:06][error]   2017-04-14T08:09:06.222+0000     Failed: error dumping metadata: error converting index (<nil>): conversion of BSON value '2' of type 'bson.Decimal128' not supported
[2017/04/14 08:09:06][error] 
[2017/04/14 08:09:06][error] Backtrace:
[2017/04/14 08:09:06][error]   /var/lib/gems/2.3.0/gems/backup-4.3.0/lib/backup/utilities.rb:199:in `run'
[2017/04/14 08:09:06][error]   /var/lib/gems/2.3.0/gems/backup-4.3.0/lib/backup/utilities.rb:217:in `block (2 levels) in <module:Helpers>'
[2017/04/14 08:09:06][error]   /var/lib/gems/2.3.0/gems/backup-4.3.0/lib/backup/database/mongodb.rb:78:in `dump!'
[2017/04/14 08:09:06][error]   /var/lib/gems/2.3.0/gems/backup-4.3.0/lib/backup/database/mongodb.rb:61:in `perform!'
[2017/04/14 08:09:06][error]   /var/lib/gems/2.3.0/gems/backup-4.3.0/lib/backup/model.rb:269:in `each'
[2017/04/14 08:09:06][error]   /var/lib/gems/2.3.0/gems/backup-4.3.0/lib/backup/model.rb:269:in `block in perform!'
[2017/04/14 08:09:06][error]   /var/lib/gems/2.3.0/gems/backup-4.3.0/lib/backup/model.rb:268:in `each'
[2017/04/14 08:09:06][error]   /var/lib/gems/2.3.0/gems/backup-4.3.0/lib/backup/model.rb:268:in `perform!'
[2017/04/14 08:09:06][error]   /var/lib/gems/2.3.0/gems/backup-4.3.0/lib/backup/cli.rb:156:in `perform'
[2017/04/14 08:09:06][error]   /var/lib/gems/2.3.0/gems/thor-0.18.1/lib/thor/command.rb:27:in `run'
[2017/04/14 08:09:06][error]   /var/lib/gems/2.3.0/gems/thor-0.18.1/lib/thor/invocation.rb:120:in `invoke_command'
[2017/04/14 08:09:06][error]   /var/lib/gems/2.3.0/gems/thor-0.18.1/lib/thor.rb:363:in `dispatch'
[2017/04/14 08:09:06][error]   /var/lib/gems/2.3.0/gems/thor-0.18.1/lib/thor/base.rb:439:in `start'
[2017/04/14 08:09:06][error]   /var/lib/gems/2.3.0/gems/backup-4.3.0/bin/backup:5:in `<top (required)>'
[2017/04/14 08:09:06][error]   /usr/local/bin/backup:22:in `load'
[2017/04/14 08:09:06][error]   /usr/local/bin/backup:22:in `<main>'
[2017/04/14 08:09:06][warn] Cleaner::Error: Cleanup Warning
[2017/04/14 08:09:06][warn]   The temporary packaging folder still exists!
[2017/04/14 08:09:06][warn]   '/root/Backup/.tmp/mongo_backup'
[2017/04/14 08:09:06][warn]   This folder may contain completed Archives and/or Database backups.
[2017/04/14 08:09:06][warn]   
[2017/04/14 08:09:06][warn]   Make sure you check these files before the next scheduled backup for
[2017/04/14 08:09:06][warn]   'MongoDB backup (mongo_backup)'
[2017/04/14 08:09:06][warn]   These files will be removed at that time!
[2017/04/14 08:09:06][info] Performing Backup for 'MongoDB backup (mongo_backup)'!
[2017/04/14 08:09:06][info] [ backup 4.3.0 : ruby 2.3.3p222 (2016-11-21 revision 56859) [x86_64-linux-gnu] ]
[2017/04/14 08:09:06][info] Database::MongoDB Started...
[2017/04/14 08:09:06][info] Running system utility 'mongodump'...
```

This is how I run the image:

```
export MONGO_CONTAINER=mongodb
docker run --name mongo-backup \
  --link $MONGO_CONTAINER:mongo \
  -e MONGODB_HOST=mongodb \
  -e INTERVAL=1d \
  -e DROPBOX_APIKEY=xxx \
  -e DROPBOX_APISECRET=xxx \
  -e DROPBOX_PATH=mongo-backups \
  --restart=always \
  -d kontena/mongo-backup
```

This are my running images:
```
docker ps -a
CONTAINER ID        IMAGE                                    COMMAND                  CREATED             STATUS              PORTS                                      NAMES
9183f3ce697e        kontena/mongo-backup                     "/bin/sh -c 'bundl..."   28 minutes ago      Up 28 minutes                                                  mongo-backup
a7a7066ebc65        jrcs/letsencrypt-nginx-proxy-companion   "/bin/bash /app/en..."   16 hours ago        Up 15 hours                                                    app-nginx-letsencrypt
1165fe0d00aa        jwilder/nginx-proxy                      "/app/docker-entry..."   16 hours ago        Up 15 hours         0.0.0.0:80->80/tcp, 0.0.0.0:443->443/tcp   app-nginx-proxy
e1e4f7470e89        naschpitz/imagemagick                    "/bin/sh -c 'bash ..."   16 hours ago        Up 15 hours         80/tcp                                     app
d207d2a91e78        mongo:3.4.1                              "/entrypoint.sh mo..."   21 hours ago        Up 15 hours         127.0.0.1:27017->27017/tcp                 mongodb
```

I managed to dump successfully inside the image using (I can also dump the mongodb outside of the docker images as I have mongo installed on the server):
`docker exec -it mongo-backup  mongodump --host mongodb --db my-db-name --archive=backup.gz --gzip`

**Comment by holms** [comment]
I this there would be more point to inherit mongodb image then to install your own mongo version in Dockerfile. That's why this happened, because you're using 3.4.1 mongo version and in Dockerfile it's 3.2 installed.

**Comment by KiwiKilian** [comment]
Didn't see that... Fixed it by using Mongo 3.2, thanks a lot!
